instrumentation/graphql/test/instrumentation/graphql/tracers/graphql_trace_test.rb

```diff
@@ -0,0 +1,292 @@
+          }
+
+          SomeOtherGraphQLAppSchema.execute('{ simpleField }')
+
+          span = spans.find { |s| s.name == 'graphql.authorized' }
+          _(span).wont_be_nil
+          _(span.attributes.to_h).must_equal(expected_attributes)
+        end
+      end
+
+      describe 'when platform_resolve_type is enabled with legacy naming' do
+        let(:config) { { enable_platform_resolve_type: true, legacy_platform_span_names: true } }
+
+        it 'traces .resolve_type' do
+          skip unless supports_authorized_and_resolved_types?
+          SomeGraphQLAppSchema.execute('{ vehicle { __typename } }')
+
+          span = spans.find { |s| s.name == 'Vehicle.resolve_type' }
+          _(span).wont_be_nil
+        end
+      end
+
+      describe 'when platform_resolve_type is enabled' do
+        let(:config) { { enable_platform_resolve_type: true } }
+
+        it 'traces .resolve_type' do
+          skip unless supports_authorized_and_resolved_types?
+          SomeGraphQLAppSchema.execute('{ vehicle { __typename } }')
+
+          span = spans.find { |s| s.name == 'graphql.resolve_type' && s.attributes['graphql.type.name'] == 'Vehicle' }
+          _(span).wont_be_nil
+        end
+
+        it 'traces .resolve_type_lazy' do
+          skip unless supports_authorized_and_resolved_types?
+          SomeGraphQLAppSchema.execute('{ vehicle { __typename } }', context: { lazy_type_resolve: true })
+
+          span = spans.find do |s|
+            s.name == 'graphql.resolve_type' &&
+              s.attributes['graphql.type.name'] == 'Vehicle' &&
+              s.attributes['graphql.lazy'] == true
+          end
+
+          _(span).wont_be_nil
+        end
+
+        it 'includes attributes' do
+          skip unless supports_authorized_and_resolved_types?
+          expected_attributes = {
+            'graphql.type.name' => 'Vehicle',
+            'graphql.lazy' => false
+          }
+
+          SomeGraphQLAppSchema.execute('{ vehicle { __typename } }')
+
+          span = spans.find { |s| s.name == 'graphql.resolve_type' }
+          _(span).wont_be_nil
+          _(span.attributes.to_h).must_equal(expected_attributes)
+        end
+      end
+
+      it 'traces validate with events' do
+        SomeGraphQLAppSchema.execute(
+          <<-GRAPHQL
+            {
+              nonExistentField
+            }
+          GRAPHQL
+        )
+        span = spans.find { |s| s.name == 'graphql.validate' }
+        event = span.events.find { |e| e.name == 'graphql.validation.error' }
+        # rubocop:disable Layout/LineLength
+        _(event.attributes['message']).must_equal(
+          "[{\"message\":\"Field 'nonExistentField' doesn't exist on type 'Query'\",\"locations\":[{\"line\":2,\"column\":15}],\"path\":[\"query\",\"nonExistentField\"],\"extensions\":{\"code\":\"undefinedField\",\"typeName\":\"Query\",\"fieldName\":\"nonExistentField\"}}]"
+        )
+        # rubocop:enable Layout/LineLength
```

Comment:
Is this not already disabled in the rubocop config?

Reply:
Unfortunately not, I'm trying to minimize the amount of changes in this PR because I'm already copy paste and shuffling a bunch of test code around.